src/__internal__/utils/helpers/events/events.test.ts

```diff
@@ -0,0 +1,330 @@
+  test("Returns true when the event type is a Space key event", () => {
+    const event = { key: " " } as KeyboardEvent;
+    const result = Events.isSpaceKey(event);
+
+    expect(result).toBeTruthy();
+  });
+
+  test("Returns false when the event type is not a Space key event", () => {
+    const event = { key: "Tab" } as KeyboardEvent;
+    const result = Events.isSpaceKey(event);
+
+    expect(result).toBeFalsy();
+  });
+});
+
+describe("isHomeKey", () => {
+  test("Returns true when the event type is a Home key event", () => {
+    const event = { key: "Home" } as KeyboardEvent;
+    const result = Events.isHomeKey(event);
+
+    expect(result).toBeTruthy();
+  });
+
+  test("Returns false when the event type is not a Home key event", () => {
+    const event = { key: "End" } as KeyboardEvent;
+    const result = Events.isHomeKey(event);
+
+    expect(result).toBeFalsy();
+  });
+});
+
+describe("isEndKey", () => {
+  test("Returns true when the event type is an End key event", () => {
+    const event = { key: "End" } as KeyboardEvent;
+    const result = Events.isEndKey(event);
+
+    expect(result).toBeTruthy();
+  });
+
+  test("Returns false when the event type is not an End key event", () => {
+    const event = { key: "Home" } as KeyboardEvent;
+    const result = Events.isEndKey(event);
+
+    expect(result).toBeFalsy();
+  });
+});
+
+describe("composedPath", () => {
+  test("Returns an empty array if there is no target", () => {
+    const event = new CustomEvent("click");
+    const result = Events.composedPath(event);
+
+    expect(result).toEqual([]);
+  });
+
+  test("Returns an empty array if target is null", () => {
+    const event = { target: null } as CustomEvent;
+    const result = Events.composedPath(event);
+
+    expect(result).toEqual([]);
+  });
+
+  test("Returns an empty array if there is no parent element", () => {
+    const event = { target: document as EventTarget } as CustomEvent;
+    const result = Events.composedPath(event);
+
+    expect(result).toEqual([]);
+  });
+
+  test("Returns the path from event.composedPath() if it is available", () => {
+    const path = Symbol("path");
+    const composedPath = jest.fn();
+    composedPath.mockReturnValue(path);
+
+    const event = new CustomEvent("click");
+    event.composedPath = composedPath;
+    const result = Events.composedPath(event);
+
+    expect(result).toBe(path);
+  });
+
+  test("Builds path from DOM elements if event.composedPath is unavailable", () => {
+    const div = document.createElement("div");
+    const ul = document.createElement("ul");
+    const li = document.createElement("li");
+
+    ul.appendChild(li);
+    div.appendChild(ul);
+
+    const path = [div, ul, li];
+    const event = ({ target: li } as unknown) as CustomEvent;
+    const result = Events.composedPath(event);
+
+    expect(result).toEqual(path);
+  });
+
+  test("Builds the path if it is not available on the element from a ReactWrapper", () => {
```

Comment:
**comment**: I wonder if we will still need our custom version of `composedPath` after Enzyme is removed. It might be worth us investigating this in the future.

What do you think about raising a ticket for this and noting its number in this test?

```suggestion
  /* TODO: <ticket-number> Investigate if `composedPath` is still required post removal of Enzyme.  */
  test("Builds the path if it is not available on the element from a ReactWrapper", () => {
```